avocado/plugins/runners/vmimage.py

```diff
@@ -0,0 +1,148 @@
+import hashlib
+import os
+import sys
+import traceback
+from multiprocessing import set_start_method
+
+from avocado.core.nrunner.app import BaseRunnerApp
+from avocado.core.nrunner.runner import BaseRunner
+from avocado.core.utils import messages
+from avocado.utils import vmimage
+
+
+class VMImageRunner(BaseRunner):
+
+    name = "vmimage"
+    description = "Runner for dependencies of type vmimage"
+
+    def run(self, runnable):
+        try:
+            yield messages.StartedMessage.get()
+
+            provider = runnable.kwargs.get("provider")
+            version = runnable.kwargs.get("version")
+            arch = runnable.kwargs.get("arch")
+
+            if not all([provider, version, arch]):
+                stderr = "Missing required parameters: provider, version, and arch"
+                yield messages.StderrMessage.get(stderr.encode())
+                yield messages.FinishedMessage.get("error")
+                return
+
+            try:
+                yield messages.StdoutMessage.get(
+                    f"Getting VM image for {provider} {version} {arch}".encode()
+                )
+
+                yield messages.StdoutMessage.get(
+                    f"Downloading VM image for: provider={provider}, version={version}, arch={arch}".encode()
+                )
+                try:
+                    from avocado.core.settings import settings
+
+                    cache_dirs = settings.as_dict().get("datadir.paths.cache_dirs")
```

Comment:
Since this runner uses `datadir.paths.cache_dirs` we need to mention it in `CONFIGURATION_USED` variable. See https://github.com/avocado-framework/avocado/blob/master/avocado/plugins/runners/avocado_instrumented.py#L35